clarifai/client/model.py

```diff
@@ -1,22 +1,168 @@
-from clarifai_grpc.grpc.api import resources_pb2, service_pb2  # noqa: F401
+import os
+import time
+from typing import Dict, List, Union
+
+from clarifai_grpc.grpc.api import resources_pb2, service_pb2
+from clarifai_grpc.grpc.api.resources_pb2 import Input
+from clarifai_grpc.grpc.api.status import status_code_pb2
+
 from clarifai.client.base import BaseClient
+from clarifai.client.lister import Lister
+from clarifai.errors import UserError
+from clarifai.utils.logging import get_logger
+from clarifai.utils.misc import BackoffIterator
 
 
-class Model(BaseClient):
+class Model(Lister, BaseClient):
   """
   Model is a class that provides access to Clarifai API endpoints related to Model information.
   Inherits from BaseClient for authentication purposes.
   """
 
-  def __init__(self, model_id: str, **kwargs):
+  def __init__(self,
+               model_id: str,
+               model_version: Dict = {'id': ""},
+               output_config: Dict = {'min_value': 0},
+               **kwargs):
     """Initializes an Model object.
     Args:
         model_id (str): The Model ID to interact with.
+        model_version (dict): The Model Version to interact with.
+        output_config (dict): The output config to interact with.
+          min_value (float): The minimum value of the prediction confidence to filter.
+          max_concepts (int): The maximum number of concepts to return.
+          select_concepts (list[Concept]): The concepts to select.
+          sample_ms (int): The number of milliseconds to sample.
         **kwargs: Additional keyword arguments to be passed to the ClarifaiAuthHelper.
     """
-    self.kwargs = {**kwargs, 'id': model_id}
+    self.kwargs = {**kwargs, 'id': model_id, 'model_version': model_version,
+                   'output_info': {'output_config': output_config}}
     self.model_info = resources_pb2.Model(**self.kwargs)
-    super().__init__(user_id=self.user_id, app_id=self.app_id)
+    self.logger = get_logger(logger_level="INFO")
+    BaseClient.__init__(self, user_id=self.user_id, app_id=self.app_id)
+    Lister.__init__(self)
+
+  def predict(self, inputs: List[Input]):
+    """Predicts the model based on the given inputs.
+    Args:
+        inputs (list[Input]): The inputs to predict, must be less than 128.
+    """
+    if len(inputs) > 128:
+      raise UserError("Too many inputs. Max is 128.")  # TODO Use Chunker for inputs len > 128
```

Comment:
> How would this work for a mutli-modal model?

We planned to add Input base class and users can directly use this function for multi model inputs.